pkg/proxy/routes.go

```diff
@@ -99,8 +99,8 @@ var (
 	// Routing rule for traffic from proxy.
 	fromProxyRule = route.Rule{
 		Priority: linux_defaults.RulePriorityFromProxyIngress,
-		Mark:     linux_defaults.MagicMarkIsProxy,
-		Mask:     linux_defaults.MagicMarkHostMask,
+		Mark:     linux_defaults.MagicMarkIngress,
+		Mask:     linux_defaults.MagicMarkProxyNoIDMask,
```

Comment:
MagicMarkProxyNoIDMask int = 0xFFFFFEFF, this patch is also trying to route egress packets while original one touches ingress only. Maybe make it clear in PR title and commit message?